README.md
# cellsyntClass #
This is a PHP class for sending SMS messages through Cellsynt's
(http://www.cellsynt.se) API. It's made to be as easy and straight forward as
possible, so anyone can implement it in their own applications.

## Usage ##
Start with including the file cellsyntClass.php in your application. Next off
create a new object of the class 'Cellsynt' with appropriate parameters,
username, password etc. Then, to actually send an SMS call the objects function
call 'sendSMS' with the parameters phone number, message and optionally expiry
date. See the quick example below.

    <?php
    require ("cellsyntClass/cellsyntClass.php");

    $myCellsynt = new Cellsynt("myuser", "sEcReT", "alpha", "Test");
    print $myCellsynt->sendSMS("0046701234567", "This is just a test SMS!");
    ?>
    
### List of avaliable parameters ###
Parameters for the class in order:

 * username (required)
 * password (required)
 * originator type (optional, default is 'alpha')
 * originator (optional, default is 'Cellsynt')
 * concatenated (optional, default is maximum, 6)
 * charset (optional, default is UTF-8)
 * type (optional, default is text (only text and flash works with this class))

Parameters for the sendSMS function in order:

 * phone number (required)
 * message (required)
 * expiry date (optional, Unix Timestamp when message should expire)

## Requirements ##
The class requires PHP5 and the PHP5 cURL module (php5-curl on Debian systems).

## Examples ##
In the examples directory are two easy to follow example on how this class can
be used. Both examples should work out of the box as is.

## Copyright ##
This class is written by Jack-Benny Persson and released under GNU GPL
version 2. 

Note that I am not in any way affiliated with Cellsynt. 


HISTORY
0.2 - 2014-08-17
    Changed properties from public to private and removed
    public keyword from the functions since this is the
    defualt anyway

0.1.1 - 2014-08-02
    Added two examples of how to use the class

0.1 - 2014-07-31
    First relase of the class

Note: For more history information, please see the git log


class.php
<?php
/*
    Copyright (C) 2014 Jack-Benny Persson <jack-benny@cyberinfo.se>

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/


/* 
    Usage example:
    $MySMS = new Cellsynt("myuser", "mYpaSS", "alpha", "Test");
    print $MySMS->sendSMS("0046701234567", "This is my SMS!");
*/

class Cellsynt
{
    public $username;
    public $password;
    public $origType;
    public $originator;
    public $phone;
    public $concat;
    public $charset;
    public $type;
    public $expiry;
    public $msg;

    public function __construct($username, $password, $origType = "alpha",
                                $originator = "Cellsynt", $concat = 6,
                                $charset = "UTF-8", $type = "text")
    {
        $this->username = $username;
        $this->password = $password;
        $this->origType = $origType;
        $this->originator = $originator;
        $this->concat = $concat;
        $this->charset = $charset;
        $this->type = $type;
    }

    public function sendSMS($phoneNr, $msg, $expiry = "")
    {
        $this->msg = $msg;
        $this->phoneNr = $phoneNr;
        $this->expiry = $expiry;
        
        $params = array ("username" => "$this->username",
                         "password" => "$this->password",
                         "destination" => "$this->phoneNr",
                         "originatortype" => "$this->origType",
                         "originator" => "$this->originator",
                         "allowconcat" => "$this->concat",
                         "charset" => "$this->charset",
                         "type" => "$this->type",
                         "expiry" => "$this->expiry",
                         "text" => "$this->msg");

        $ch = curl_init();
        curl_setopt($ch, CURLOPT_HEADER, 0);
        curl_setopt($ch, CURLOPT_URL, "https://se-1.cellsynt.net/sms.php");
        curl_setopt($ch, CURLOPT_POST, 1);
        curl_setopt($ch, CURLOPT_POSTFIELDS, (http_build_query($params)));
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
        $status = curl_exec($ch);
        curl_close($ch);
        return $status;
    }
}


?>


examples/webpage.php
<?php
/* 
    An example of what the cellsyntClass can be used for.
    Here is a simple webpage from which one can send SMS-message by specifing
    phone number, text message and sender information.
*/

// the class is required for this to work
require ("../cellsyntClass.php");

// start the HTML
print <<<_START_HTML_
<!DOCTYPE html>
<head>
<title>Example SMS webpage</title>
<meta charset="UTF-8" />
</head>

<body>
<h1>Send SMS-message</h1>

<form action="$_SERVER[PHP_SELF]" method="post">
Phone number:<input type="text" name="phone" size="18" maxlength="18"><br/>
Message:<br/>
<textarea name="message" cols="70" rows="15"></textarea><br/>
Sender:<input type="text" name="sender" size="11" maxlength="11"><br/>
<input type="submit" value="Send SMS"><br/>
</form>
_START_HTML_;

// check if the form has been submitted in which case we send the SMS
if((isset($_POST['phone'])) || (isset($_POST['message'])) ||
    (isset($_POST['sender'])))
{
    // make variables of the _POST array
    $phone = $_POST['phone'];
    $message = $_POST['message'];
    $sender = $_POST['sender'];

    // instantiate class with $sender and hardcoded user and password
    $MySMS = new Cellsynt("myuser", "sEcReT", "alpha", "$sender");
    // call the sendSMS function and print the status-message on the webpage
    print "<p>" . $MySMS->sendSMS($phone, $message) . "</p>\n";
}

?>

</body>
</html>


examples/sms_as_args.php
<?php
/* 
    A small example of what can be done with the cellcentClass.
    Here is a script that let's a user send SMS-messages from the command-line
    by giving two arguments to the script, the first one is the phone number,
    the second is the text message to be sent.
    Login information and originator text is set as static values here.
*/

// the class is required...
require ("../cellsyntClass.php");

// username and password information for the Cellsynt account and 
// the sender/originator
$username = "myuser";
$password = "sEcReT";
$sender = "From CLI";

// run a sanity check
sanitycheck();

// if we got past the sanity checks above, instantiate the class
$cliSMS = new Cellsynt($username, $password, "alpha", $sender);

// put the arguments into variables for readability
$phone = $argv[1];
$message = $argv[2];

// call the sendSMS function with the phone number and message
$status = $cliSMS->sendSMS($phone, $message);

// check the status of the message and print the result
checkstatus();

// if we got to down here, something was wrong, exit with status code 1
exit(1);


function checkstatus()
{
    global $status;
    // check for an OK message at the start of the line, in that case everything
    // was fine
    if (preg_match("/^OK/", $status))
    {
        print "Message was sent successfully. Status message reads:\n";
        print $status . "\n";
        exit(0);
    }
    // if there were no OK message, something didn't work
    else
    {
        print "Something didn't work out as expected. Status message reads:\n";
        print $status . "\n";
        exit(1);
    }

}

function sanitycheck()
{
    global $argv;
    // check if user entered two arguments
    if (!isset($argv[1]) || !isset($argv[2])) 
    {
        print "Usage: $argv[0] <phone nr> <'quoted text message'>\n";
        exit(1);
    }

    // check if the phone number looks valid
    if (!preg_match("/\d{7,16}/", $argv[1]))
    {
        print "That dosen't look like a valid phone number\n";
        exit(1);
    }
}

?>


cellsyntClass.php
<?php
/*
    Copyright (C) 2014 Jack-Benny Persson <jack-benny@cyberinfo.se>

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/


/* 
    Version 0.2

    Usage example:
    $MySMS = new Cellsynt("myuser", "mYpaSS", "alpha", "Test");
    print $MySMS->sendSMS("0046701234567", "This is my SMS!");
*/

class Cellsynt
{
    private $username;
    private $password;
    private $origType;
    private $originator;
    private $phone;
    private $concat;
    private $charset;
    private $type;
    private $expiry;
    private $msg;

    function __construct($username, $password, $origType = "alpha",
                                $originator = "Cellsynt", $concat = 6,
                                $charset = "UTF-8", $type = "text")
    {
        $this->username = $username;
        $this->password = $password;
        $this->origType = $origType;
        $this->originator = $originator;
        $this->concat = $concat;
        $this->charset = $charset;
        $this->type = $type;
    }

    function sendSMS($phoneNr, $msg, $expiry = "")
    {
        $this->msg = $msg;
        $this->phoneNr = $phoneNr;
        $this->expiry = $expiry;
        
        $params = array ("username" => "$this->username",
                         "password" => "$this->password",
                         "destination" => "$this->phoneNr",
                         "originatortype" => "$this->origType",
                         "originator" => "$this->originator",
                         "allowconcat" => "$this->concat",
                         "charset" => "$this->charset",
                         "type" => "$this->type",
                         "expiry" => "$this->expiry",
                         "text" => "$this->msg");

        $ch = curl_init();
        curl_setopt($ch, CURLOPT_HEADER, 0);
        curl_setopt($ch, CURLOPT_URL, "https://se-1.cellsynt.net/sms.php");
        curl_setopt($ch, CURLOPT_POST, 1);
        curl_setopt($ch, CURLOPT_POSTFIELDS, (http_build_query($params)));
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
        $status = curl_exec($ch);
        curl_close($ch);
        return $status;
    }
}


?>
